client/main.js

```diff
@@ -0,0 +1,66 @@
+const data = document.getElementById("form");
+const publish = document.getElementById("publish");
+
+for (const el of data.children) {
+  if (el.classList.contains("empty")) {
+    el.textContent = el.dataset.placeholder;
+  }
+}
+
+publish.addEventListener('click', postSecret)
+
+// REQUESTS
+
+async function getAll(secrets) {
+  try {
+    const response = await fetch(`http://localhost:3000/${secrets}`);
+    const data = await response.json();
+    return data;
+  } catch (err) {
+    console.warn(err);
+  }
+}
+
+async function getById(secrets, id) {
+  try {
+    const response = await fetch(`http://localhost:3000/${secrets}/${id}`);
+    const data = await response.json();
+    return data;
+  } catch (err) {
+    console.warn(err);
+  }
+}
+
+async function postSecret(e) {
+  e.preventDefault();
+
+  const inputData = []
+
+  for (const el of data.children) {
```

Comment:
I cant see where data is defined. I am guessing you are trying to posting a form here, so you can target the whole form and then do 
`form.onsubmit = postSecret()`
then on line 49 :

Reply:
that should do what you are trying to do. 
Ref: the assignment request.js and modal.js